src/main/java/com/nirima/jenkins/plugins/docker/DockerContainerWatchdog.java

```diff
@@ -109,6 +109,11 @@ protected boolean stopAndRemoveContainer(DockerAPI dockerApi, Logger logger, Str
     
     @Override
     protected void execute(TaskListener listener) throws IOException, InterruptedException {
+        if (!JenkinsUtils.getSystemPropertyBoolean(DockerContainerWatchdog.class.getName()+".enabled", true)) {
```

Comment:
I notice that we've got a change in strategy here - this code is reading the system property every time the code runs, whereas the other times we read the system properties for this kind of information, we only do so at classloading time.
This seems to be inconsistent, and makes me suspect that there may not be a good reason for the inconsistency.

i.e. Perhaps the other calls to `JenkinsUtils.getSystemProperty`Xxx should be made at runtime instead of classload time, so that the values can be adjusted while the Jenkins server is running.
...well, all the values that _can_ be adjusted while the Jenkins server is running, that is - I see that `getRecurrencePeriod()` is documented as a one-off call, but `GRACE_DURATION_FOR_CONTAINERS_TO_START_WITHOUT_NODE_ATTACHED` could be re-read every time.

I think we either need to make all the properties on-startup-only _or_ (preferably) make them all "changeable at runtime" (where possible).  Personally I like the latter option.